lib/network.go

```diff
@@ -2509,11 +2592,17 @@ func (msg *MsgDeSoBlock) EncodeBlockVersion1(preSignature bool) ([]byte, error)
 	return data, nil
 }
 
+func (msg *MsgDeSoBlock) EncodeBlockVersion2(preSignature bool) ([]byte, error) {
+	return msg.EncodeBlockCommmon(preSignature)
+}
+
 func (msg *MsgDeSoBlock) ToBytes(preSignature bool) ([]byte, error) {
 	if msg.Header.Version == HeaderVersion0 {
 		return msg.EncodeBlockVersion0(preSignature)
 	} else if msg.Header.Version == HeaderVersion1 {
 		return msg.EncodeBlockVersion1(preSignature)
+	} else if msg.Header.Version == HeaderVersion2 {
```

Comment:
maybe we switch this to a switch statement? whenever I see if-else-if-else-if all checking the same fields, I generally opt for the switch statement.